src/database/anime.ts

```diff
@@ -0,0 +1,156 @@
+import Database from 'better-sqlite3';
+import ANNClient from '../structures/ANNClient.js';
+import AnisongClient, { AnisongData } from '../structures/AnisongClient.js';
+import AnilistClient from '../structures/AnilistClient.js';
+
+import config from '../config.js';
+import { CronJob } from 'cron';
+import { start } from 'repl';
+
+const db = new Database('./database/anime.db', {
+    fileMustExist: false,
+    readonly: false,
+});
+db.pragma('journal_mode=WAL');
+export default class AnimeData {
+    private ANNClient: ANNClient
+    private AnisongClient: AnisongClient
+    private AnilistClient: AnilistClient
+    constructor() {
+        this.intialize();
+        this.ANNClient = new ANNClient();
+        this.AnisongClient = new AnisongClient();
+        this.AnilistClient = new AnilistClient();
+    }
+    public intialize(): void {
+        db.prepare(
+            'CREATE TABLE IF NOT EXISTS ann (annId INTEGER PRIMARY KEY, gid INTEGER, type TEXT, name TEXT, precision TEXT, vintage TEXT, hasAnilist BOOLEAN, hasAnisong BOOLEAN)'
+        ).run();
+        db.prepare(
+            'CREATE TABLE IF NOT EXISTS anilistmedia (anilistMediaId INTEGER PRIMARY KEY, annId INTEGER, name TEXT)'
+        ).run();
+        db.prepare(
+            'CREATE TABLE IF NOT EXISTS anisong (anisongId INTEGER PRIMARY KEY, annId INTEGER, anilistMediaId INTEGER, url TEXT, songType TEXT, anisongType TEXT, animeEng TEXT, animeJap TEXT, songName TEXT, animeType TEXT)'
+        ).run();
+        const data: any = db.prepare('SELECT * FROM ann LIMIT 1').get();
+        if (!data) {
+            this.pullAllFromANN();
+        }
+        const job1 = new CronJob('0 0 0 * * *', this.pullRecentFromANN, null, true, 'America/New_York');
```

Comment:
Should day be 0 here?

Reply:
node-cron uses (second minute hour dayofmonth month dayofweek).
0 0 0 * * * should be daily at 12am EST
0 0 0 * * 1 should be Weekly, on Monday , at 12am EST